src/Equinox/Decider.fs

```diff
@@ -108,6 +108,39 @@ type Decider<'event, 'state>(inner : DeciderCore<'event, 'state>) =
         let inline decide' c ct = task { let! r, es = Async.StartImmediateAsTask(decide c, ct) in return struct (r, Seq.ofList es) }
         return! inner.TransactExAsync(decide = decide', mapResult = mapResult, ?load = load, ?attempts = attempts, ct = ct) |> Async.AwaitTaskCorrect }
 
+    /// 1. Invoke the supplied <c>Task</c> returning <c>interpret</c> function with the present state
+    /// 2. (if events yielded) Attempt to sync the yielded events to the stream.
+    ///    (Restarts up to <c>maxAttempts</c> times with updated state per attempt, throwing <c>MaxResyncsExhaustedException</c> on failure of final attempt.)
+    /// 3. Uses <c>render</c> to generate a 'view from the persisted final state
+    member _.TransactTask(interpret : 'state -> CancellationToken -> Task<'event list>, render : 'state -> 'view, ?load, ?attempts, ?ct) : Task<'view> =  task {
```

Comment:
Have you experimented with other signatures? e.g. would calling it `TransactAsync` work or does it mess up intellisense? (I can see the reasoning about giving specific names - we can always add aliases later but you can't undo an overloaded name as easily)
is the `list` -> `seq` hop still valuable (IIRC in the async case, it mattered somewhere at some time, but that might have been an old compiler/VS toolset on net45 a long time ago)

Reply:
Also wondering whether exposing a lower level TransactExAsync (or `inner`) might allow this overall need to be served by adding extension methods/type augmentations locally vs requiring every impl to be a PR and release?

Reply:
Also wondering are any of these signatures useful in dotnet-templates?

Reply:
> Have you experimented with other signatures?

I'll admit that I didn't, I've been knee deep in OCaml (which doesn't support any overloading whatsoever) for the past few days so it didn't even occur to me to overload.

If it is possible would that be your preference?

Reply:
> is the list -> seq hop still valuable

I'm not sure to be honest, I assumed it was there intentionally.

Reply:
The root issue here is that we need to know if there are events pretty much straight away in https://github.com/jet/equinox/blob/master/src/Equinox/Core.fs#L44
That was once boring F# async code, and `list` made sense - pattern matching etc
Recently for C#/DeciderCore, I made that be `Task<_, 'res * 'e seq>` as a LCD between C# `yield return`s (which IIRC can't be used to build arrays neatly?) and the `list`s from F#
I suspect that changing the `list` to `seq` will trigger some compiler errors in F# consumption in some relatively exotic cases. (I think I validated that, but not 100% sure)

Reply:
Isn't a list already a `seq`? can't we just pass the list?

Reply:
Yes, it is. There are exotic cases (e.g. `>>` etc) wherein things sometimes break down though. They are definitely pretty rare and/or arguably shortcomings in type inference and/or type generalization